lib/src/command/run.dart

```diff
@@ -126,41 +118,15 @@ class RunCommand extends PubCommand {
       args = argResults.rest.skip(1).toList();
     }
 
-    String snapshotPath(String command) => p.join(
-          entrypoint.cachePath,
-          'bin',
-          package,
-          '$command.dart.snapshot.dart2',
-        );
-
-    // If snapshot exists, we strive to avoid using [entrypoint.packageGraph]
-    // because this will load additional files. Instead we just run with the
-    // snapshot. Note. that `pub get|upgrade` will purge snapshots.
-    var snapshotExists = fileExists(snapshotPath(command));
-
-    // Don't ever compile snapshots for mutable packages, since their code may
-    // change later on. Don't check if this the case if a snapshot already
-    // exists.
-    var useSnapshot = snapshotExists ||
-        (package != entrypoint.root.name &&
-            !entrypoint.packageGraph.isPackageMutable(package));
+    final experiments = argResults['enable-experiment'] as List;
+    final vmArgs = <String>[
+      if (experiments.isNotEmpty) "--enable_experiment=${experiments.join(',')}"
```

Comment:
ditto

Reply:
Seeing this logic three times, wondering if it's worth a utility 🤔

Reply:
I made a function!